### PR TITLE
Yiz/get user

### PR DESCRIPTION
**Pull request by sleepyzzzzzz**
## Changes made in this PR
add GetUser implementation + specs in grpc

## Corresponding noted Ruby branch
https://github.com/ParamountDataManagement/paramount-api/pull/1967

## Implementation Checklist
Please check following check list to ensure the tasks are considered and completed in this branch.

#### API Design
- [x] Defined gRPC API in proto file
- [x] regenerated go code using `make proto_generate` command

#### API implementation
- [x] Implemented API in corresponding gRPC server
- [x] Added tests in gRPC layer

#### Domain Service and DB query Implementation
- [ ] Added new query in dbstore layer
- [ ] Added tests in db store layer
                                                                                       
- [ ] Implemented functionality in Domain Service Layer
- [ ] Added test in Domain Service Layer

- [ ] Created a new branch in paramount-api, with comments there indicating the functionality in Ruby that's re-implemented in Golang

#### Postgres Layer - identify and potentially add db migration to create index for fast querying.
- [ ] Verified and potentially created index for the newly added query.

### CODE REVIEW

File: apis/grpcservice/grpcusers/usersGRPCServer_test.go [hunk 1]
```diff
@@ -514,6 +514,71 @@ truncate table receivers cascade;
 		})
 	})
 
+	DescribeTable("GetUser", func(
+		inputGetUserRequest *usersv1.GetUserRequest,
+		expectedErrorOccurred bool,
+		expectedResponse *usersv1.User,
```
Comment (niki-joanne) [inline apis/grpcservice/grpcusers/usersGRPCServer_test.go:520]: Minor nitpick: Can we switch the order of the last two parameters? 
Just seems better for readability if the "test" parameter comes last in the argument list

Comment (sleepyzzzzzz) [reply apis/grpcservice/grpcusers/usersGRPCServer_test.go:528]: fix

File: apis/pdm/users/v1/users_service.proto [hunk 2]
```diff
@@ -129,6 +130,14 @@ message ConfirmUserResponse{
   string redirection_url = 1;
 }
 
+message GetUserRequest{
+  int32 user_id = 1;
```
Comment (lhan-the-real) [inline apis/pdm/users/v1/users_service.proto:134]: change to `id`. 

This is already in the scope of `GetUser` so we can just use id to indicate it.

File: apis/grpcservice/grpcusers/usersGRPCServer_test.go [hunk 3]
```diff
@@ -523,6 +523,71 @@ truncate table receivers cascade;
...
+				Email:     testUserEmail,
+				Password:  password,
+				AccountID: testAccount.ID,
+			},
+		)
+		Expect(err).To(BeNil())
+
+		if !expectedErrorOccurred {
+			inputGetUserRequest.UserId = user.ID
+		}
```
Comment (lhan-the-real) [inline apis/grpcservice/grpcusers/usersGRPCServer_test.go:566]: Why we need to check this?

File: apis/pdm/users/v1/users_service.proto (hunk 2 above)
Comment (sleepyzzzzzz) [reply apis/pdm/users/v1/users_service.proto]: fix

File: apis/grpcservice/grpcusers/usersGRPCServer_test.go (hunk 3 above)
Comment (sleepyzzzzzz) [reply apis/grpcservice/grpcusers/usersGRPCServer_test.go:566]: It's not a check. 
CreateUser does not take in the id and id is auto-generated by the SQL. So for the input, there is no way to know what's the id for the newly created user and I wrote this to assign the value

Comment (lhan-the-real) [reply apis/grpcservice/grpcusers/usersGRPCServer_test.go]: Ah... I see. If you've merged from main, I changed the logic to allow insert user with a given id. This is to allow for inserting the same user info same user id in postgres as what exists in mysql.

So you can take advantage of that and clean up the logic here.

**Review by lhan-the-real — APPROVED** [review]

File: apis/grpcservice/grpcusers/usersGRPCServer_test.go (hunk 3 above)
Comment (sleepyzzzzzz) [reply apis/grpcservice/grpcusers/usersGRPCServer_test.go]: fix

**Pull request merged**